SnapML_models/SnapMLEfficientFormer.py

```diff
@@ -0,0 +1,516 @@
+                                                  f'{self.__class__.__name__} '
+            if self.init_cfg is not None:
+                ckpt_path = self.init_cfg['checkpoint']
+            elif pretrained is not None:
+                ckpt_path = pretrained
+
+            ckpt = _load_checkpoint(
+                ckpt_path, logger=logger, map_location='cpu')
+            if 'state_dict' in ckpt:
+                _state_dict = ckpt['state_dict']
+            elif 'model' in ckpt:
+                _state_dict = ckpt['model']
+            else:
+                _state_dict = ckpt
+
+            state_dict = _state_dict
+            missing_keys, unexpected_keys = \
+                self.load_state_dict(state_dict, False)
+
+    def forward_tokens(self, x):
+        outs = []
+        for idx, block in enumerate(self.network):
+            x = block(x)
+        return x
+
+    def forward(self, x):
+        x = self.patch_embed(x)
+        x = self.forward_tokens(x)
+        if self.fork_feat:
+            # output features of four stages for dense prediction
+            return x
+        x = self.norm(x)
+        cls_out = self.head(x.mean(0)), self.dist_head(x.mean(0))
+        cls_out = (cls_out[0] + cls_out[1]) / 2
+        cls_out = cls_out.unsqueeze(0).softmax(1)
+        # for image classification
+        return cls_out
+
+@register_model
+def SnapML_efficientformer_l0(pretrained=False, **kwargs):
+    model = SnapML_EfficientFormer(
+        layers=EfficientFormer.EfficientFormer_depth['l0'],
+        embed_dims=EfficientFormer.EfficientFormer_width['l0'],
+        downsamples=[True, True, True, True],
+        vit_num=2,
+        **kwargs)
+    model.default_cfg = EfficientFormer._cfg(crop_pct=0.9)
+    return model
+
+
+@register_model
+def SnapML_efficientformer_l1(pretrained=False, **kwargs):
+    model = SnapML_EfficientFormer(
+        layers=EfficientFormer.EfficientFormer_depth['l1'],
+        embed_dims=EfficientFormer.EfficientFormer_width['l1'],
+        downsamples=[True, True, True, True],
+        vit_num=1,
+        **kwargs)
+    model.default_cfg = EfficientFormer._cfg(crop_pct=0.9)
+    return model
+
+
+@register_model
+def SnapML_efficientformer_l3(pretrained=False, **kwargs):
+    model = SnapML_EfficientFormer(
+        layers=EfficientFormer.EfficientFormer_depth['l3'],
+        embed_dims=EfficientFormer.EfficientFormer_width['l3'],
+        downsamples=[True, True, True, True],
+        vit_num=4,
+        **kwargs)
+    model.default_cfg = EfficientFormer._cfg(crop_pct=0.9)
+    return model
+
+
+@register_model
+def SnapML_efficientformer_l7(pretrained=False, **kwargs):
+    model = SnapML_EfficientFormer(
+        layers=EfficientFormer.EfficientFormer_depth['l7'],
+        embed_dims=EfficientFormer.EfficientFormer_width['l7'],
+        downsamples=[True, True, True, True],
+        vit_num=8,
+        **kwargs)
+    model.default_cfg = EfficientFormer._cfg(crop_pct=0.9)
+    return model
```

Comment:
Maybe we can also remove `SnapML_efficientformer_l3` & `SnapML_efficientformer_l7` since not used?